rules-tests/Autodiscovery/Rector/Class_/MoveValueObjectsToValueObjectDirectoryRector/MoveValueObjectsToValueObjectDirectoryRectorTest.php

```diff
@@ -7,11 +7,14 @@
 use Iterator;
 use Rector\FileSystemRector\ValueObject\AddedFileWithContent;
 use Rector\Testing\PHPUnit\AbstractRectorTestCase;
+use Rector\Testing\PHPUnit\PlatformAgnosticAssertions;
 use Symplify\SmartFileSystem\SmartFileInfo;
 use Symplify\SmartFileSystem\SmartFileSystem;
 
 final class MoveValueObjectsToValueObjectDirectoryRectorTest extends AbstractRectorTestCase
 {
+    use PlatformAgnosticAssertions;
+
     /**
```

Comment:
> Having trait in multiple files might lead to memory-lock.

what do you mean?


> How can we remove this burden from each test class?

IMO its a great feature to have a opt-in for this functionality, instead of blindly have it anywhere

the alternative could be to put the methods directly into e.g. `Rector\Testing\PHPUnit\AbstractTestCase` but then all tests, even those which not need this platform agnostic parts, would be affected and might get side-effect

Reply:
> what do you mean?

It's 1:M problem. If someone forget the trait here, it would fail. That's a memory-lock "when A then B" problem.

> IMO its a great feature to have a opt-in for this functionality, instead of blindly have it anywhere

The goal is to abstract OS dependeny in tests. Otherwise we have slash methods and endline correctoin spamming all over the tests. Ideally no-one should notice :) 

> the alternative could be to put the methods directly into e.g. Rector\Testing\PHPUnit\AbstractTestCase but then all tests, even those which not need this platform agnostic parts, would be affected and might get side-effect

:+1: This would be easier to maintain as we have single class to extend.

Reply:
Atm we would need the trait in something arround 10-15 test classes. The big majority already work without further relaxing of the assertion.

I am fine with moving it to AbstractRectorTest if you prefer

Reply:
sorry for arguing. will adjust per your decision now.